pkg/registry/file/vulnsummarystorage.go

```diff
@@ -0,0 +1,240 @@
+	if err != nil {
+		if errors.Is(err, afero.ErrFileNotFound) {
+			return nil, storage.NewKeyNotFoundError(path, 0)
+		}
+		logger.L().Ctx(ctx).Error("read file failed", helpers.Error(err), helpers.String("key", path))
+		return nil, err
+	}
+	err = json.Unmarshal(b, &vulnManifestSummary)
+	if err != nil {
+		logger.L().Ctx(ctx).Error("json unmarshal failed", helpers.Error(err), helpers.String("key", path))
+		return nil, err
+	}
+	if _, exist := vulnManifestSummary.GetLabels()["kubescape.io/workload-namespace"]; !exist {
+		return nil, storage.NewKeyNotFoundError(path, 0)
+	}
+	return &vulnManifestSummary, nil
+}
+
+// add uni test
+func initNamespaceVulnSummary(key string) softwarecomposition.VulnerabilitySummary {
+	namespace := filepath.Base(key)
+	var vulnSummaryByNamespace softwarecomposition.VulnerabilitySummary
+	vulnSummaryByNamespace.APIVersion = "spdx.softwarecomposition.kubescape.io/v1beta1"
+	vulnSummaryByNamespace.Kind = "VulnerabilitySummary"
+	vulnSummaryByNamespace.Labels = map[string]string{"kubescape.io/workload-namespace": namespace}
+	vulnSummaryByNamespace.Annotations = map[string]string{"kubescape.io/status": ""}
+	vulnSummaryByNamespace.Name = namespace
+	vulnSummaryByNamespace.Namespace = namespace
+	vulnSummaryByNamespace.CreationTimestamp.Time = time.Now()
+
+	return vulnSummaryByNamespace
+}
+
+func updateVulnCounters(aggregatedCounters *softwarecomposition.VulnerabilityCounters, counters *softwarecomposition.VulnerabilityCounters) {
+	aggregatedCounters.All += counters.All
+	aggregatedCounters.Relevant += counters.Relevant
+}
+
+func updateSeverities(aggregatedSeverities *softwarecomposition.SeveritySummary, severities *softwarecomposition.SeveritySummary) {
+	updateVulnCounters(&aggregatedSeverities.Critical, &severities.Critical)
+	updateVulnCounters(&aggregatedSeverities.High, &severities.High)
+	updateVulnCounters(&aggregatedSeverities.Medium, &severities.Medium)
+	updateVulnCounters(&aggregatedSeverities.Low, &severities.Low)
+	updateVulnCounters(&aggregatedSeverities.Negligible, &severities.Negligible)
+	updateVulnCounters(&aggregatedSeverities.Unknown, &severities.Unknown)
+}
+
+func updateFullVulnSumm(fullVulnSumm *softwarecomposition.VulnerabilitySummary, vulnManifestSumm *softwarecomposition.VulnerabilityManifestSummary) {
+	updateSeverities(&fullVulnSumm.Spec.Severities, &vulnManifestSumm.Spec.Severities)
+	if vulnManifestSumm.Spec.Vulnerabilities.ImageVulnerabilitiesObj != (softwarecomposition.VulnerabilitiesObjScope{}) {
+		fullVulnSumm.Spec.WorkloadVulnerabilitiesObj = append(fullVulnSumm.Spec.WorkloadVulnerabilitiesObj, vulnManifestSumm.Spec.Vulnerabilities.ImageVulnerabilitiesObj)
+	}
+	if vulnManifestSumm.Spec.Vulnerabilities.WorkloadVulnerabilitiesObj != (softwarecomposition.VulnerabilitiesObjScope{}) {
+		fullVulnSumm.Spec.WorkloadVulnerabilitiesObj = append(fullVulnSumm.Spec.WorkloadVulnerabilitiesObj, vulnManifestSumm.Spec.Vulnerabilities.WorkloadVulnerabilitiesObj)
+	}
```

Comment:
issue: it is hard to understand what this function does and why

What are we doing here? Are we merging manifest summaries? Which ones? _Why_ might we need to merge them?

Reply:
please take a look on the docs reference